File: testdata/python_small/mylib/db/connection.py
"""Database connection management."""

from mylib.config import Config

_connection = None


def get_connection():
    """Get or create a database connection."""
    global _connection
    if _connection is None:
        _connection = _create_connection(Config.DATABASE_URL)
    return _connection


def close_connection():
    """Close the active database connection."""
    global _connection
    if _connection is not None:
        _connection = None


def _create_connection(url):
    """Create a new database connection from URL."""
    return {"url": url, "connected": True}

File: testdata/ts_small/src/db/connection.ts
/** Database connection management. */

import pg from "pg";
import { getDbConfig } from "../config";
import type { DbConfig } from "../types";

let pool: any = null;

export function getConnection(): any {
  if (!pool) {
    const config = getDbConfig();
    pool = createPool(config);
  }
  return pool;
}

function createPool(config: DbConfig): any {
  return { config, connected: true };
}

export function closeConnection(): void {
  if (pool) {
    pool = null;
  }
}
